design/specs/add-formula-column.md

```diff
@@ -0,0 +1,180 @@
+
+![](https://share.balsamiq.com/c/sqqrcNeq3CVQaUtZiBG9ov.png)
+
+#### Scenario 1.f - Add a boolean type formula column
+
+The boolean type formula works on columns with boolean data type. There can be different parameters depending on what the formula is. For instance, there is `condition` here in the example below.
+
+![](https://share.balsamiq.com/c/nBHVJCNiZ2PgJMrn1wdGcL.png)
+
+#### Scenario 1.g - Add a regular expression type formula column
+
+The regular expression type formula works on columns with any data type. There can be different parameters depending on what the formula is. It basically is to determine if there are any patterns in the data of a column. 
+
+![](/https://share.balsamiq.com/c/dNYLKaCt4oJAgZ9qE7rFy4.png)
+
+#### Scenario 1.h - Add a list type formula column
+
+The list type formula works on columns with list data type. There can be different parameters depending on what the formula is.
+
+![](https://share.balsamiq.com/c/wnDL3cvcR9R3cgPwR3Co2.png)
+
+### Scenario 2 - The column added as a parameter gets deleted 
+
+When the column added to the formula gets deleted by user, the formula column will be retained in the error state. So the data in the formula column with have `Error` as value.
+
+### Scenario 3 - Change in the data type of the column added as a parameter
+
+When the parameter column data type is changed to an incompatible data type, the formula column will be retained in the error state. So the data in the formula column with have `Error` as value.
+
+### Scenario 4 - Add invalid values to the formula parameters
+
+The user cannot add invalid columns as they won't be present in the list but they might add invalid values. If the values are invalid the data in the formula column with have `Error` as value.
+
+### Scenario 5 - Edit a formula column 
+
+The formula can be edited anytime user wants. So there can be two scenarios on how users goes about changing the formula.
+
+Case 1 - The user changes the parameters.
+So here, user will be only given options for compatible columns. And if they want it to change it to a value, they can even do that. But an invalid value will again show `Error` in the column. 
+
+Case 2 - The user changes the selected formula.
+Once the user selects a different formula parameter values are not retained and they see empty parameters and an empty column again. To see the value they will have to add compatible params again.
+
+### Scenario 6 - Click outside the inspector
+
+#### Scenario 6.a - Clicks outside before adding name or selecting formula
+
+If the users click outside the inspector before even touching the formula settings that is adding name or selecting formula, there won't be any empty column added. And there would be absolutely no change in the explorer.
+
```

Comment:
I believe we discussed changing this so that all columns added remain until explicitly deleted by the user.

Reply:
You are right. Will change